ascend-sdk/src/main/java/ai/evolv/ascend/android/ExecutionQueue.java

```diff
@@ -0,0 +1,41 @@
+package ai.evolv.ascend.android;
+
+import com.google.gson.JsonArray;
+
+import java.util.LinkedList;
+
+import ai.evolv.ascend.android.exceptions.AscendKeyError;
+import timber.log.Timber;
+
+class ExecutionQueue {
+
+    private final LinkedList<Execution> queue;
```

Comment:
LinkedLists are not thread-safe. You should use a ConcurrentLinkedQueue or implement locking.

Reply:
Ok, ill adopt the ConcurrentLinkedQueue